app/overrides/spree/order_deface.rb

```diff
@@ -0,0 +1,28 @@
+script = '
+<% intent_secrets = @order.payments.map do |payment|
+  next unless payment.intent_client_key
+  {
+    intent_key: payment.intent_client_key,
+    pk_key: payment.payment_method.preferred_publishable_key
+  }
+end.last %>
+
+<% if intent_secrets %>
+  <script type="text/javascript" src="https://js.stripe.com/v3/"></script>
+  <script>
+
+    function confirmCardPaymentResponseHandler(response) {
+      $.post("/stripe/intents/handle", { response: response, order: "<%= @order.number %>", authenticity_token: "<%= form_authenticity_token %>" })
+    };
+
+    var stripeElements = Stripe("<%= intent_secrets[:pk_key] %>");
+    stripeElements.confirmCardPayment("<%=intent_secrets[:intent_key]%>").then(function(result) {
```

Comment:
@chhga this is done on the checkout payment step, why is it here?

Reply:
@damianlegawiec well, it's actually not. This method fire 3ds confirmation modal. We cannot call it before order completed, just because on automatic payment capture it charges a client immediately.

Reply:
@chhga we need to call it before order completion - let's do it on the confirm step when the user clicks `save and continue`

If confirm step isn't available then the payment step it is

Reply:
@damianlegawiec we're unable to do it, just because in spree flow we receive payment intent only on payment `.purchase` or `.authorize` action. It happens as a before_action for order completion.